### DISCUSSION TITLE
[MRTK3] Set reprojection method

### DISCUSSION THREAD

**Discussion by pajama** [microsoft/MixedRealityToolkit-Unity#11123] [Q&A]
I would like to use Automatic Planar Reprojection. I'm wondering where that setting has moved in MRTK3. Thanks!

**Comment by Zee2** [comment]
I'm not sure if we support specifying reprojection type over OpenXR. @keveleigh ?

We might bind to the https://docs.unity3d.com/ScriptReference/XR.XRDisplaySubsystem-reprojectionMode.html API, but this doesn't specify depth-aware vs planar, just rotational vs full-6dof.

**Comment by pajama** [comment]
This at least compiles. Will check to see if it works! I'm not sure what creates the ViewConfiguration in mrtk3.

```
using System;
using System.Linq;
using Microsoft.MixedReality.OpenXR;
using UnityEngine;

public class ReprojectionController : MonoBehaviour
{
    public ReprojectionMode ReprojectionMode = ReprojectionMode.PlanarFromDepth;

    private ReprojectionSettings _reprojectionSettings = default;
    private void OnPostRender()
    {
        _reprojectionSettings.ReprojectionMode = ReprojectionMode;
        foreach (ViewConfiguration viewConfiguration in ViewConfiguration.EnabledViewConfigurations)
        {
            if (viewConfiguration.IsActive && viewConfiguration.SupportedReprojectionModes.Contains(ReprojectionMode))
            {
                viewConfiguration.SetReprojectionSettings(_reprojectionSettings);
            }
        }

    }
}

```

**Reply by pajama** [reply]
I'm not really sure how to validate that this is working actually because there doesn't seem to be an option for disabling the reprojection override in the windows device portal.

**Reply by keveleigh** [reply]
This looks like the correct API and reasonable logic to me!

**Reply by pajama** [reply]
@keveleigh it doesn't seem to work unfortunately. No errors, but definitely not switching the reprojection method.

**Reply by yl-msft** [reply]
Is there any reason you were using "OnPoseRender" function call?  Not sure if we tested in this scenario,  just curious.

You can also look at the "Rendering" scene in this BasicSample unity project on GitHub, which allow you to iterate through different projection mode and observe different effects.

Another curious question, @pajama ,  are you running this on HoloLens2?  or VR device?   What depth format did you setup for your unity setting?

**Reply by pajama** [reply]
I was just copying an example from mrtk2. This is using mrtk3 on HL. Please link to an example script if there is one for mrtk3. Thanks so much!

**Reply by satyapoojasama** [reply]
@pajama - This issue has nothing to do with MRTK in specific. It is using Unity OpenXR API to set reprojection settings. From my testing, there seems to be no issues with setting target reprojection with this API.

Can you try using this Basic Sample "Rendering" scene and script to see if it works for you? https://github.com/microsoft/OpenXR-Unity-MixedReality-Samples/blob/main/BasicSample/Assets/Rendering/Scripts/RenderingSettings.cs#L74
 Also, the only difference between your snippet above and this script is where the logic is run - "OnPostRender()" vs "Update()".
